dotCMS/src/main/java/com/dotcms/jobs/business/api/JobQueueManagerAPI.java

```diff
@@ -0,0 +1,118 @@
+package com.dotcms.jobs.business.api;
+
+import com.dotcms.jobs.business.error.CircuitBreaker;
+import com.dotcms.jobs.business.error.JobCancellationException;
+import com.dotcms.jobs.business.error.ProcessorNotFoundException;
+import com.dotcms.jobs.business.error.RetryStrategy;
+import com.dotcms.jobs.business.job.Job;
+import com.dotcms.jobs.business.processor.JobProcessor;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+/**
+ * Defines the contract for interacting with the job queue system. This interface provides methods
+ * for managing jobs, processors, and the overall state of the job queue.
+ */
+public interface JobQueueManagerAPI extends AutoCloseable {
+
+    /**
+     * Starts the job queue manager, initializing the thread pool for job processing.
+     */
+    void start();
+
+    /**
+     * Checks if the JobQueueManager has been started.
+     *
+     * @return {@code true} if the JobQueueManager has been started, {@code false} otherwise.
+     */
+    boolean isStarted();
+
+    /**
+     * Waits for the JobQueueManager to start up.
+     *
+     * @param timeout The maximum time to wait.
+     * @param unit    The time unit of the timeout argument.
+     * @return {@code true} if the JobQueueManager has started, {@code false} if the waiting time
+     * elapsed before the JobQueueManager started.
+     * @throws InterruptedException if the current thread is interrupted while waiting.
+     */
+    boolean awaitStart(long timeout, TimeUnit unit) throws InterruptedException;
```

Comment:
this shouldnt has a future or callback version such as
void awaitStart(long timeout, TimeUnit unit, () -> " Logger.debug(this, "Not is running")) throws InterruptedException;